cmux-tui/scripts/verify_published_manifest.py

```
@@ -0,0 +1,197 @@
+            raise ManifestError(f"{source} missing required artifact {artifact}")
+
+
+def _read_url(url: str) -> bytes:
+    # The post-publish latest check uses a ``?verify=<commit>`` cache-buster,
+    # so queries are allowed. Fragments are rejected because HTTP never sends
+    # them to the origin and they cannot identify the verified document.
+    try:
+        parsed = urlsplit(url)
+        hostname = parsed.hostname
+    except ValueError as error:
+        raise ManifestError("manifest URL is malformed") from error
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)